File: array/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i=0;
        int sum = 0;
        int length = INT32_MAX;

        for (int j=0; j<nums.size();++j){
            sum+=nums[j];

            while(sum>=target){
                length = length<(j-i+1)?length:(j-i+1);
                sum-=nums[i++];
            }
        }

        return length==INT32_MAX?0:length;
    }
};

File: binary-search/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        sum = 0
        length = float('inf')

        for right in range(len(nums)):
            sum += nums[right]

            while sum>=target:
                length = min(length, right - left + 1)
                sum -= nums[left]
                left+=1

        if length == float('inf'):
            return 0
        else:
            return length